### ISSUE TITLE
pip install fails on Linux, Python 3.10

### ISSUE THREAD

**Issue by jmcelroy01** [InsightSoftwareConsortium/ITK#3587]
### Description

When running pip install itk, I get the following output: 

```
[358/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_matlab_filewrite.cxx.o
      [359/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_numeric_traits.cxx.o
      [360/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_matops.cxx.o
      [361/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_real_polynomial.cxx.o
      [362/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_rational.cxx.o
      [363/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_bignum.cxx.o
      FAILED: Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_bignum.cxx.o
      /usr/bin/c++ -DVXL_WARN_DEPRECATED -DVXL_WARN_DEPRECATED_ONCE -I/tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/v3p/netlib -I/tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/vcl -I/tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/core -I/tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKb/Modules/ThirdParty/VNL/src/vxl/v3p/netlib -I/tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKb/Modules/ThirdParty/VNL/src/vxl/vcl -I/tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKb/Modules/ThirdParty/VNL/src/vxl/core -I/tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/core/vnl -mtune=native -march=corei7 -Wall -Wcast-align -Wdisabled-optimization -Wextra -Wformat=2 -Winvalid-pch -Wno-format-nonliteral -Wpointer-arith -Wshadow -Wunused -Wwrite-strings -funit-at-a-time -Wno-strict-overflow -Wno-deprecated -Wno-invalid-offsetof -Woverloaded-virtual -Wstrict-null-sentinel  -w  -fPIC -O3 -DNDEBUG -fPIC   -Wno-undefined-var-template -msse2 -std=c++11 -MD -MT Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_bignum.cxx.o -MF Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_bignum.cxx.o.d -o Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_bignum.cxx.o -c /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/core/vnl/vnl_bignum.cxx
      /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/core/vnl/vnl_bignum.cxx: In member function ‘vnl_bignum::operator float() const’:
      /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/core/vnl/vnl_bignum.cxx:716:37: error: ‘numeric_limits’ is not a member of ‘std’
        716 |   if (this->is_infinity()) f = std::numeric_limits<float>::infinity();
            |                                     ^~~~~~~~~~~~~~
      /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/core/vnl/vnl_bignum.cxx:716:52: error: expected primary-expression before ‘float’
        716 |   if (this->is_infinity()) f = std::numeric_limits<float>::infinity();
            |                                                    ^~~~~
      /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/core/vnl/vnl_bignum.cxx: In member function ‘vnl_bignum::operator double() const’:
      /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/core/vnl/vnl_bignum.cxx:727:37: error: ‘numeric_limits’ is not a member of ‘std’
        727 |   if (this->is_infinity()) d = std::numeric_limits<double>::infinity();
            |                                     ^~~~~~~~~~~~~~
      /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/core/vnl/vnl_bignum.cxx:727:52: error: expected primary-expression before ‘double’
        727 |   if (this->is_infinity()) d = std::numeric_limits<double>::infinity();
            |                                                    ^~~~~~
      /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/core/vnl/vnl_bignum.cxx: In member function ‘vnl_bignum::operator long double() const’:
      /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/core/vnl/vnl_bignum.cxx:738:37: error: ‘numeric_limits’ is not a member of ‘std’
        738 |   if (this->is_infinity()) d = std::numeric_limits<long double>::infinity();
            |                                     ^~~~~~~~~~~~~~
      /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKs/Modules/ThirdParty/VNL/src/vxl/core/vnl/vnl_bignum.cxx:738:52: error: expected primary-expression before ‘long’
        738 |   if (this->is_infinity()) d = std::numeric_limits<long double>::infinity();
            |                                                    ^~~~
      At global scope:
      cc1plus: note: unrecognized command-line option ‘-Wno-undefined-var-template’ may have been intended to silence earlier diagnostics
      [364/5515] Building C object Modules/ThirdParty/VNL/src/vxl/v3p/netlib/CMakeFiles/itkv3p_netlib.dir/opt/lbfgsb.c.o
      [365/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_bignum_traits.cxx.o
      [366/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_decnum_traits.cxx.o
      [367/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_int_matrix.cxx.o
      [368/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_rational_traits.cxx.o
      [369/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_least_squares_cost_function.cxx.o
      [370/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_least_squares_function.cxx.o
      [371/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_real_npolynomial.cxx.o
      [372/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_crs_index.cxx.o
      [373/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_cost_function.cxx.o
      [374/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_linear_system.cxx.o
      [375/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_nonlinear_minimizer.cxx.o
      [376/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_sparse_matrix_linear_system.cxx.o
      [377/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_fastops.cxx.o
      [378/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_rotation_matrix.cxx.o
      [379/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_sparse_lst_sqr_function.cxx.o
      [380/5515] Building CXX object Modules/ThirdParty/VNL/src/vxl/core/vnl/CMakeFiles/itkvnl.dir/vnl_decnum.cxx.o
      ninja: build stopped: subcommand failed.
      FAILED: ITKp/src/ITK-stamp/ITK-build /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKp/src/ITK-stamp/ITK-build
      cd /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKb && /home/user/.local/lib/python3.10/site-packages/cmake/data/bin/cmake --build . && /home/user/.local/lib/python3.10/site-packages/cmake/data/bin/cmake -E touch /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build/ITKp/src/ITK-stamp/ITK-build
      ninja: build stopped: subcommand failed.
      Traceback (most recent call last):
        File "/usr/lib/python3.10/site-packages/skbuild/setuptools_wrap.py", line 595, in setup
          cmkr.make(make_args, env=env)
        File "/usr/lib/python3.10/site-packages/skbuild/cmaker.py", line 609, in make
          raise SKBuildError(
      
      An error occurred while building with CMake.
        Command:
          cmake --build . --target install --config Release --
        Source directory:
          /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec
        Working directory:
          /tmp/pip-install-elp4aiw5/itk_208682f461dd4624bae34af40ec0d0ec/_skbuild/linux-x86_64-3.10/cmake-build
      Please see CMake's output for more information.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> itk

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

### Steps to Reproduce
1. I am running OpenSUSE Tumbleweed, Python 3.10.6, cmake 3.24.1. 
2. After pip installing SimpleITK, cmake, cmake-common, ninja, and installing doxygen from the OpenSUSE repos, I was able to get further in the install process than before. Pip has also been updated. 

### Expected behavior
For pip install to work. 

Is there a library or version of something related to the DVXL error I could try to install?

**Comment by dzenanz** [comment]
It looks like `vnl_bignum.h` does `#include <limits>` so I don't know what causes `error: ‘numeric_limits’ is not a member of ‘std’`. This looks more like a question for OpenSUSE ITK package maintainer.

But, aren't pip-installable packages pre-compiled?

@hjmjohnson do you have an idea?

**Comment by thewtex** [comment]
Python 3.10 binaries are available in the 5.3 pre-release. They can be installed via:

```
pip install --upgrade --pre itk
```
